peer_protocol/peer.go

```diff
@@ -1,20 +1,56 @@
 package peer_protocol
 
 import (
+	"../config"
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	"math/rand"
 	"net"
+	"time"
 )
 
 type PeerConnection struct {
-	Conn         net.Conn
-	PeerID       []byte
-	InfoHash     []byte
-	remotePeerID []byte
+	Conn           net.Conn
+	RemotePeerID   []byte
+	Active         bool
+	Config         *config.PeerConfig
+	AmChoking      bool
+	PeerChoking    bool
+	PeerInterested bool
+	bitField       []byte
 }
 
-func (p PeerConnection) createMessageBuffer(mt byte, payload []byte) []byte {
+func NewPeer(conn net.Conn, config *config.PeerConfig) PeerConnection {
+	return PeerConnection{
+		Conn:        conn,
+		Config:      config,
+		PeerChoking: true,
+	}
+}
+
+func (p *PeerConnection) readSocket(len uint32, timeout int) ([]byte, error) {
+	buf := make([]byte, len)
+	err := p.Conn.SetReadDeadline(time.Now().Add(time.Second * time.Duration(timeout)))
+	if err == nil {
+		_, err := p.Conn.Read(buf)
+		return buf, err
+	} else {
+		return buf, err
+	}
+}
+
+func (p *PeerConnection) Disconnect() {
+	err := p.Conn.Close()
+	if err != nil {
+		fmt.Printf("failed closing connection with client %s", p.Conn.RemoteAddr().String())
```

Comment:
Consider to use log instead of fmt, I think it might be helpful in case you'll later want to write your logs to a file

Reply:
will do